### DISCUSSION TITLE
Is there any documentation about how Cosmos is built?

### DISCUSSION THREAD

**Discussion by alexpdp7** [jart/cosmopolitan#928] [Q&A] [answered]
(Basically that, I'd love to try getting Emacs 29.1 built.)

**Comment by pkulchenko** [comment, marked as answer]
Check @ahgamut's https://github.com/ahgamut/superconfigure repository. It has bunch of binaries, including emacs-28.2, so you can check its configuration on how it's done.

**Reply by alexpdp7** [reply]
Ah, that's cool, thanks. Esp. Emacs also has some info in the Cosmopolitan documentation. I'll take a look.

I'm quite curious to see Cosmos' build too. One of the cool things I wanted to see if someone bootstrapping a Linux distro on WASM, but the same thing on Cosmopolitan seems much more interesting to me now. Not all the daemons, services, etc. Just the userland stuff...

**Reply by alexpdp7** [reply]
In case anyone is curious, I tried https://github.com/ahgamut/superconfigure/pull/6 , but that's not enough to build 29.1.

**Reply by alexpdp7** [reply]
ahgamut has been kind enough to help me get 29.1 built- there's more info in the PR. I'm seeing that superconfigure's build is different from the Cosmos one (e.g. Cosmos includes `emacsclient`, but superconfigure doesn't), happy to send patches to Cosmos if that's published somewhere.
